commands/kubernetes.go

```diff
@@ -1941,26 +2101,72 @@ func clusterByIDorName(kube do.KubernetesService, idOrName string) (*do.Kubernet
 // clusterIDize attempts to make a cluster ID/name string be a cluster ID.
 // use this as opposed to `clusterByIDorName` if you just care about getting
 // a cluster ID and don't need the cluster object itself
-func clusterIDize(kube do.KubernetesService, idOrName string) (string, error) {
-	if looksLikeUUID(idOrName) {
-		return idOrName, nil
-	}
-	clusters, err := kube.List()
-	if err != nil {
-		return "", err
+func clusterIDize(c *CmdConfig, idOrName string) (string, error) {
+	return iDize(c, idOrName, "cluster")
+}
+
+// iDize attempts to make a resource ID/name string be a resource ID.
+// use this if you just care about getting a resource ID and don't need the object itself
+func iDize(c *CmdConfig, resourceIDOrName string, resType string) (string, error) {
+	if looksLikeUUID(resourceIDOrName) {
+		return resourceIDOrName, nil
 	}
 	var ids []string
-	for _, c := range clusters {
-		if c.Name == idOrName {
-			id := c.ID
-			ids = append(ids, id)
+
+	switch resType {
+	case "volume":
+		volumes, err := c.Volumes().List()
+		if err != nil {
+			return "", err
+		}
+
+		for _, v := range volumes {
+			if v.Name == resourceIDOrName {
```

Comment:
For volumes, if we want to allow for specifying names, we'll need to ensure both the name and the region match. Volume names are only unique per region.

```
$ doctl compute volume ls --format ID,Name,Region | grep test-volume
9a53df21-6245-11eb-b05f-0a58ac14542f    test-volume                                 nyc1
a2848aa7-6245-11eb-89c9-0a58ac1475e8    test-volume                                 nyc3
a615354a-6245-11eb-965a-0a58ac14c196    test-volume                                 lon1
aa14619f-6245-11eb-9863-0a58ac14d18c    test-volume                                 fra1
```

Reply:
Okay, since the volume will be created in the cluster's region, I can just check against that, right?

Reply:
Think that should do the trick!

Reply:
Thanks @andrewsomething! Updated it.